File: LeetCode - April/numberOfIsland.java
class Solution {
    public int numIslands(char[][] grid) {
        if(grid == null || grid.length == 0) return 0;
		
        int numIslands=0;
		
        for(int i=0; i<grid.length; i++){
            for(int j=0; j<grid[i].length; j++){
                if(grid[i][j] == '1'){
                    numIslands +=dfs(grid, i, j);
                }
            }
        }
        return numIslands;
    }
    
    int dfs(char[][] array, int i, int j){
        if(i<0 || i>=array.length || j<0 || j>=array[i].length || array[i][j]=='0') return 0;
        
        array[i][j]='0'; 
        
        dfs(array, i+1, j); 
        dfs(array, i-1, j);
        dfs(array, i, j+1);  
        dfs(array, i,j-1);  
        return 1;
    }
}

File: LeetCode - April/numberOfIsland.py
class Solution:
    def numIslands(self, grid: List[List[str]]) -> int:
        if not grid: 
            return 0
        rows , cols , numberOfIslands = len(grid) , len(grid[0]) , 0
        
        def findIsland(grid , x , y):
            if x < 0 or x >=rows or y < 0 or y >= cols or grid[x][y] == "0":
                return 0
            grid[x][y] = "0"
            findIsland(grid , x + 1 , y)
            findIsland(grid , x - 1 , y)
            findIsland(grid , x , y + 1)
            findIsland(grid , x , y - 1)
            return 1
        
        for i in range(rows):
            for j in range(cols):
                if grid[i][j] == "1":
                    numberOfIslands += findIsland(grid , i , j)
        return numberOfIslands